model/messaging/spans.yaml

```diff
@@ -118,6 +118,21 @@ groups:
           Message [correlation Id](https://www.rabbitmq.com/tutorials/tutorial-six-java#correlation-id) property.
       - ref: messaging.message.body.size
         requirement_level: opt_in
+      - ref: messaging.destination.name
+        note: |
+          In RabbitMQ, the destination is defined by an *exchange*, a *routing key* and for consumers, a *queue*.
+
+          `messaging.destination.name` SHOULD be set to:
+
+          - **Producers**: `{exchange}:{routing key}` when both values are present and non-empty.
+          Otherwise: `amq.default` when the default exchange is used and no routing key is provided
```

Comment:
can there be an `{exchange}` but not a `{routing key}`, or can there be a `{routing key}` but not a `{exchange}`? it may be helpful to be explicit about fallback options (if any), in particular, do you drop the `:` or not when only one of them is present, here's an example where we cover these fallbacks: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/database-spans.md#name

Reply:
Yeah they can be empty like that AFAIK. I adapted it a bit, hopefully it is clearer now.